src/components/organisms/FileTreePane/FileTreePane.tsx

```diff
@@ -668,6 +668,15 @@ const FileTreePane = () => {
     [loadLastFolderOnStartup, setFolder, recentFolders]
   );
 
+  // called from main thread to avoid running setRootFolder in main thread
+  useEffect(() => {
+    ipcRenderer.on('set-root-folder', (_, data) => {
```

Comment:
please add logic to remove the event listener using `return` like we did on the `useEffect` below this one to avoid memory leaks